src/corelibs/U2Formats/src/FastaFormat.cpp

```diff
@@ -317,19 +317,24 @@ static void writeHeaderToFile(IOAdapterWriter &writer, const QString &sequenceNa
     writer.write(os, FastaFormat::FASTA_HEADER_START_SYMBOL + sequenceName + '\n');
 }
 
-// TODO: a recommended width for FASTA lines is 80 chars to be readable in text editors.
-static constexpr qint64 SAVE_LINE_LEN = 1024 * 1024;
+/** Length of the FASTA sequence line. Using 80 symbols to make FASTA files readable in text editors. */
+static constexpr qint64 FASTA_SEQUENCE_LINE_LENGTH = 80;
 
 static void saveSequenceObject(IOAdapterWriter &writer, const U2SequenceObject *sequence, U2OpStatus &os) {
     writeHeaderToFile(writer, sequence->getSequenceName(), os);
     CHECK_OP(os, );
     qint64 sequenceLength = sequence->getSequenceLength();
-    for (int i = 0; i < sequenceLength; i += SAVE_LINE_LEN) {
-        qint64 chunkSize = qMin(SAVE_LINE_LEN, sequenceLength - i);
-        QByteArray chunkContent = sequence->getSequenceData(U2Region(i, chunkSize), os);
+    for (int i = 0; i < sequenceLength; i += FASTA_SEQUENCE_LINE_LENGTH) {
+        qint64 chunkSize = qMin(FASTA_SEQUENCE_LINE_LENGTH, sequenceLength - i);
+        U2Region region(i, chunkSize);
+        QByteArray chunkContent = sequence->getSequenceData(region, os);
         CHECK_OP(os, );
         writer.write(os, QString::fromLatin1(chunkContent));
         CHECK_OP(os, );
+        if (region.endPos() < sequenceLength) {
+            writer.write(os, "\n");
+            CHECK_OP(os, );
+        }
     }
     writer.write(os, "\n");
```

Comment:
Now you have two endline characters in the end of document. I think, we can remove this line